newdle/client/src/components/NewdlesParticipating.js

```diff
@@ -0,0 +1,99 @@
+import flask from 'flask-urls.macro';
+import {Container, Icon, Label, Placeholder} from 'semantic-ui-react';
+import {useHistory} from 'react-router';
+import {serializeDate, toMoment} from '../util/date';
+import client from '../client';
+import {usePageTitle} from '../util/hooks';
+import styles from './MyNewdles.module.scss';
+
+export default function NewdlesParticipating() {
+  const [participations, loading] = client.useBackend(() => client.getNewdlesParticipating(), []);
+  usePageTitle("Newdles I'm participating in");
+
+  let content;
+  if (loading || participations === null) {
+    content = (
+      <>
+        <Placeholder className={styles.newdle} />
+        <Placeholder className={styles.newdle} />
+        <Placeholder className={styles.newdle} />
+      </>
+    );
+  } else if (participations.length === 0) {
+    content = (
+      <div className={styles['no-newdle-container']}>
+        {/* eslint-disable-next-line jsx-a11y/accessible-emoji */}
+        <h2>You are not part of any newdles yet.</h2>
+      </div>
+    );
+  } else {
+    content = participations.map(participation => (
+      <NewdleParticipation
+        key={participation.newdle.id}
+        newdle={participation.newdle}
+        answers={participation.answers}
+        participant_code={participation.code}
+      />
+    ));
+  }
+
+  return (
+    <Container text className={styles.newdles}>
+      {content}
+    </Container>
+  );
+}
+
+function NewdleParticipation({
+  newdle: {code, title, duration, final_dt: finalDT, timezone, timeslots},
+  answers,
+  participant_code,
+}) {
+  const history = useHistory();
+  const startTime = finalDT && serializeDate(finalDT, 'HH:mm');
+  const endTime = finalDT && serializeDate(toMoment(finalDT).add(duration, 'm'), 'HH:mm');
+  const url = flask`newdle`({code, participant_code});
+  const slotsChosen = timeslots.filter(timeslot =>
+    ['available', 'ifneedbe'].includes(answers[timeslot])
+  );
+
+  return (
+    <div className={styles.newdle} onClick={() => history.push(url)}>
+      <h3 className={styles.title}>
+        <a href={url} onClick={evt => evt.preventDefault()}>
+          {title}
+        </a>
+        <Label color={finalDT ? 'blue' : 'green'} size="tiny" className={styles.state}>
+          {finalDT ? 'finished' : 'ongoing'}
+        </Label>
+      </h3>
+      {finalDT && (
+        <div className={styles.info}>
+          <span>
+            <Icon name="calendar alternate outline" />
+            <label>{serializeDate(finalDT, 'dddd, D MMMM')}</label>
+          </span>
+          <span>
+            <Icon name="clock outline" />
+            <label>
+              {`${startTime} - ${endTime}`} ({timezone})
+            </label>
+          </span>
+        </div>
+      )}
+      <div className={styles.participants}>
+        <Icon name="clock outline" />
+        <label>
+          <em>
+            {finalDT
+              ? 'This newdle has finished'
+              : !slotsChosen.length
```

Comment:
This check isn't correct; if you do not choose any slots because you are unavailable, it does not mean it's still awaiting a reply.

Reply:
Nice catch. I skipped that part because I had the idea of treating unavailable responses with a different message.
Should I just show "You replied with X timeslot(s)" in any case, independently of the type of response?

Reply:
"You are not available" or something similar could work as well

Reply:
Before resolving this conversion, I added unavailable responses as part of "You've replied with X timeslot(s)", which should simplify the logic.

Reply:
That sounds like a good solution!